packages/@aws-cdk/core/lib/resource.ts

```diff
@@ -14,6 +15,28 @@ export interface IResource extends IConstruct {
    * The stack in which this resource is defined.
    */
   readonly stack: Stack;
+
+  /**
+   * The AWS account ID that this resource belongs to.
+   * For resources that are created and managed by the CDK
+   * (generally, those created by creating new class instances like Role, Bucket, etc.)
+   * this is always the same as the account of the stack they belong to;
+   * however, for imported resources
+   * (those obtained from static methods like fromRoleArn, fromBucketName, etc.)
+   * that might be different than the stack they were imported into.
+   */
+  readonly account: IEnvComponent;
```

Comment:
Can you wrap these up another level? 

`environment: Env` (or `env: Env`), which contains two `EnvComponent`s?

Reply:
Except we already have an interface called `Environment` with `account` and `region` properties.... won't it be confusing to add `Env` to the mix?

Reply:
Done (the new interface is called `ResourceEnvironment`).